src/manifest.rs

```diff
@@ -353,40 +349,43 @@ pub fn store_local(path: &Path, bytes: &[u8]) -> Result<()> {
 
 #[cfg(test)]
 mod test {
-    use ring::signature::Ed25519KeyPair;
-    use ring::test::rand::FixedSliceRandom;
-    use untrusted::Input;
+    use sodiumoxide::crypto::hash::sha256;
+    use sodiumoxide::crypto::sign::ed25519;
 
-    use config::PublicKey;
     use error::Error;
-    use super::{Entry, Manifest, Sha256, parse_entry};
+    use super::{Entry, Manifest, parse_entry};
     use version::Version;
 
-    fn get_test_key_pair() -> Ed25519KeyPair {
+    fn get_test_key_pair() -> (ed25519::PublicKey, ed25519::SecretKey) {
         // Produce the keypair from the same 32 bytes each time in the tests,
         // so they are deterministic. From this seed, the following key is
         // generated:
-        // Secret key: MFMCAQEwBQYDK2VwBCIEIHRlc3Qta2V5LXZlcnktc2VjdXJpdHktc3Vja
-        // C1zYWZloSMDIQCXQPbwnZ+Ihe9Y9t5k/vCRqr50HnkaXbKyKCX2ZAfb2Q==
+        // Secret key: SECRET+dGVzdC1rZXktdmVyeS1zZWN1cml0eS1zdWNoLXN
+        //             hZmWXQPbwnZ+Ihe9Y9t5k/vCRqr50HnkaXbKyKCX2ZAfb2
+        //             ZdA9vCdn4iF71j23mT+8JGqvnQeeRpdsrIoJfZkB9vZ
         // Public key: l0D28J2fiIXvWPbeZP7wkaq+dB55Gl2ysigl9mQH29k=
-        let seed = b"test-key-very-security-such-safe";
-        let rng = FixedSliceRandom { bytes: &seed[..] };
-        let pkcs8_bytes = Ed25519KeyPair::generate_pkcs8(&rng).unwrap();
-        Ed25519KeyPair::from_pkcs8(Input::from(&pkcs8_bytes)).unwrap()
+        let seed = ed25519::Seed(*b"test-key-very-security-such-safe");
+        ed25519::keypair_from_seed(&seed)
     }
 
-    fn get_test_public_key() -> PublicKey {
-        PublicKey::from_pair(&get_test_key_pair())
+    fn get_test_public_key() -> ed25519::PublicKey {
+        let (pk, _sk) = get_test_key_pair();
+        pk
+    }
+
+    fn get_test_secret_key() -> ed25519::SecretKey {
+        let (_pk, sk) = get_test_key_pair();
+        sk
     }
 
     /// A sequence of 32 bytes that I don't want to repeat everywhere.
-    fn get_test_sha256() -> Sha256 {
+    fn get_test_sha256() -> sha256::Digest {
         const TEST_SHA256: [u8; 32] = [
             0x96, 0x41, 0xa4, 0x9d, 0x02, 0xe9, 0x0c, 0xbb, 0x62, 0x13, 0xf2,
             0x02, 0xfb, 0x63, 0x2d, 0xa7, 0x0c, 0xdc, 0x59, 0x07, 0x3d, 0x42,
             0x28, 0x3c, 0xfc, 0xdc, 0x1d, 0x78, 0x64, 0x54, 0xf1, 0x7f
         ];
-        Sha256(TEST_SHA256)
+        sha256::Digest(TEST_SHA256)
```

Comment:
The default hashing algorithm of `sodium_oxide` is `SHA512` (`crypto::hash()`)
However, in the upstream version of `libsodium` the default is actually `Blake2b` , which
confuses me a little bit.

and the default signing algorithm is `ed25519`

https://sodiumoxide.github.io/sodiumoxide/sodiumoxide/crypto/sign/index.html

so we could just use the `crypto::sign` funciton


I'm not sure if it's better to use the `libsodium` defaults, or expicitly state the
algorithms we're using.  Don't have a very strong opinion on it.

Reply:
I think explicitly stating them is the only option; if the defaults change in a future libsodium release then everything would break. Changing these algorithms would require a file format change (which is why the manifest starts with a version number, to enable future upgrades while retaining backwards compatibility).